app/models/user_favourite.rb

```diff
@@ -0,0 +1,53 @@
+class UserFavourite < ApplicationRecord
+  include PushUpdatesMinimal
+
+  LIMIT_PER_TABLE = 20
+  
+  belongs_to :favourite, polymorphic: true
+  belongs_to :user
+  
+  after_create :apply_limit
+
+  # Live update rules
+  after_save :push_changes
+  after_destroy :push_changes
+  push_targets do |user_favourite|
+    Channel.private_channels_for(user_favourite.user, STOCK_APP)
+  end
+  
+  def apply_limit
+    #
+    # Delete non-persistent favorites > 20 
+    #
+    UserFavourite
+      .where(user_id: user_id, favourite_type: favourite_type, persistent: false)
+      .order('updated_at')
+      .offset(LIMIT_PER_TABLE)
+      .destroy_all
+  end
+  
+  class << self
+    #
+    # Mark a record as a favourite
+    #
+    def add_user_favourite(record, persistent: false, user: User.current_user)
+      return unless user.present?
+      
+      UserFavourite.where(
+        user: user,
+        favourite: record,
+        persistent: persistent
+      ).first_or_initialize do |fav|
+        fav.save  if fav.new_record?
+        fav.touch if fav.persisted?
```

Comment:
This may fire 2 updates in some cases, so touching the updated_at attribute manually and then saving, should fix this.

```suggestion
        fav.updated_at = Time.now
        fav.save
```